api/src/controllers/userController.js

```diff
@@ -254,17 +254,38 @@ const getUserById = (req, res) => {
 // @route   GET /api/user/profile/:userID
 // @access  Public
 const getUserProfileByUserID = async (req, res) => {
-  const id = req.params.userID
-  await db.User.findOne({ where: { userID: id } })
-    .then((profile) => {
-      if (profile) {
-        res.json(profile)
-      } else {
-        res.status(404)
-        throw new Error('Profile not found')
-      }
-    })
-    .catch((err) => res.json({ error: err.message }).status(400))
+  try {
+    const id = req.params.userID
+
+    // fetching userID for community members
+    // desc: in community member the userId is availabe not userID
+    const user = await db.User.findOne({ where: { id }, attriburtes: ['userID'] })
+
+    let userID
+    if (user?.userID) {
```

Comment:
no need to check userID as it will be there for all localauth and cognito from now on only old record did not had it.